### PR TITLE
MNT: update Shadow init signature

### PR DESCRIPTION
**Pull request by tacaswell**
## PR summary

Fixes typing bug from me merging #25389 

The CI on that PR was from before we merge the typing so this was not checked.

### CODE REVIEW

File: lib/matplotlib/patches.pyi [hunk 1]
```diff
@@ -74,7 +74,7 @@ class Patch(artist.Artist):
 
 class Shadow(Patch):
     patch: Patch
-    def __init__(self, patch: Patch, ox: float, oy: float, **kwargs) -> None: ...
+    def __init__(self, patch: Patch, ox: float, oy: float, *, shade: float **kwargs) -> None: ...
```
Comment (rcomer) [inline lib/matplotlib/patches.pyi:77]:
```suggestion
    def __init__(self, patch: Patch, ox: float, oy: float, *, shade: float, **kwargs) -> None: ...
```

File: lib/matplotlib/patches.pyi [hunk 2]
```diff
@@ -74,7 +74,7 @@ class Patch(artist.Artist):
 
 class Shadow(Patch):
     patch: Patch
-    def __init__(self, patch: Patch, ox: float, oy: float, **kwargs) -> None: ...
+    def __init__(self, patch: Patch, ox: float, oy: float, *, shade: float, **kwargs) -> None: ...
```
Comment (rcomer) [inline lib/matplotlib/patches.pyi:77]:
```suggestion
    def __init__(self, patch: Patch, ox: float, oy: float, *, shade: float = …, **kwargs) -> None: ...
```
?

Comment (rcomer) [reply lib/matplotlib/patches.pyi:77]: That `…` doesn’t look right 😕

File: lib/matplotlib/patches.pyi [hunk 3]
```diff
@@ -74,7 +74,7 @@ class Patch(artist.Artist):
 
 class Shadow(Patch):
     patch: Patch
-    def __init__(self, patch: Patch, ox: float, oy: float, **kwargs) -> None: ...
+    def __init__(self, patch: Patch, ox: float, oy: float, *, shade: float = …, **kwargs) -> None: ...
```
Comment (rcomer) [inline lib/matplotlib/patches.pyi:77]:
```suggestion
    def __init__(self, patch: Patch, ox: float, oy: float, *, shade: float = ..., **kwargs) -> None: ...
```

Sorry I think my tablet was trying to be clever on the last one.

Comment (tacaswell) [reply lib/matplotlib/patches.pyi]: I thought those ... looked too close together.....

File: lib/matplotlib/patches.pyi (hunk 2 above)
Comment (tacaswell) [reply lib/matplotlib/patches.pyi]: I thought this was a comment about `...` being used for "there is a default" in the stub files.

**Review by rcomer — APPROVED** [review]
It passes 🥳

**Comment by tacaswell** [comment]
I'm going to squash this down to one commit and we can not talk about the amount of tries it took .....

**Comment by tacaswell** [comment]
This has passed mypy and the stubs should not be used any where else so I'm going to merge before all of CI is done to un-break CI for all other PRs.

**Pull request merged**